index_documents.go

```diff
@@ -177,13 +177,15 @@ func (i Index) GetDocuments(request *DocumentsQuery, resp *DocumentsResult) erro
 	req := internalRequest{
 		endpoint:            "/indexes/" + i.UID + "/documents",
 		method:              http.MethodGet,
+		contentType:         contentTypeJSON,
 		withRequest:         nil,
 		withResponse:        resp,
-		withQueryParams:     map[string]string{},
+		withQueryParams:     nil,
 		acceptedStatusCodes: []int{http.StatusOK},
 		functionName:        "GetDocuments",
 	}
-	if request != nil {
+	if request != nil && len(request.Filter) == 0 {
```

Comment:
You did not defined a `else` clause, is that even possible to happen?

Reply:
Yes, in the default case the `internalRequest` is set at the beginning of the method, avoid both if and is executed normally.

Reply:
So you should not add the first `request != nil` :)

Reply:
> So you should not add the first request != nil

Not sure to understand well but in Go I had to check the pointer if unfortunately, I try to access a field of the request without checking if the request exists, I will encounter a panic.

But what I could do if you think is better is put to the `request != nil` in a upper level alone and the two other if inside.